src/react-imgix.jsx

```diff
@@ -124,25 +124,28 @@ function buildSrc({
       srcSet = `${dpr1} 1x, ${dpr2} 2x, ${dpr3} 3x, ${dpr4} 4x, ${dpr5} 5x`;
     } else {
       let showARWarning = false;
+      const { width, height, ...urlParams } = srcOptions;
+      const constructedUrl = constructUrl(rawSrc, urlParams);
+
+      const aspectRatioDecimal = parseAspectRatio(aspectRatio);
+      if (aspectRatio != null && aspectRatioDecimal === false) {
+        // false indicates invalid
+        showARWarning = true;
+      }
+
       const buildSrcSetPair = targetWidth => {
-        let urlParams = {
-          ...srcOptions,
-          width: targetWidth
-        };
-        const aspectRatioDecimal = parseAspectRatio(aspectRatio);
-        if (aspectRatio != null && aspectRatioDecimal === false) {
-          // false indicates invalid
-          showARWarning = true;
+        let url = constructedUrl;
+        if (height) {
+          url = `${url}&h=${height}`;
         }
-        if (
-          !srcOptions.height &&
+        else if (
           aspectRatioDecimal &&
           aspectRatioDecimal > 0
         ) {
-          urlParams.height = Math.ceil(targetWidth / aspectRatioDecimal);
+          url = `${url}&h=${Math.ceil(targetWidth / aspectRatioDecimal)}`;
```

Comment:
Optional: maybe there's more performance to be gained from doing this concatenation (along with L139) in one go with L148? (the template string in the return statement). Something like 

```js
let h;

// ...

h = Math.ceil(targetWidth / aspectRatioDecimal);

// ...

return `${url}&h={h}&w=${targetWidth} ${targetWidth}w`; 
```

Reply:
In that case we might get `&h=&w=...`. Is that acceptable?

Reply:
Never mind. I changed the behaviour a little bit 🙂